presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -82,6 +102,16 @@ public static JsonGenerator createJsonGenerator(JsonFactory factory, SliceOutput
         return factory.createGenerator((OutputStream) output);
     }
 
+    public static String getTrimmedJsonForErrorMessage(String json)
```

Comment:
`trim` is a misleading word here. What about `truncateIfNecessaryForErrorMessage`?